### DISCUSSION TITLE
Allow setting Annotations and MessageHeaders

### DISCUSSION THREAD

**Discussion by hae-andrew-young** [rabbitmq/rabbitmq-stream-dotnet-client#376] [Q&A]
### Is your feature request related to a problem? Please describe.

These are `internal` properties at the moment and we would like to be able to set them to put metadata about the payload in there.

### Describe the solution you'd like

see above.

### Describe alternatives you've considered

We could set them in `ApplicationProperties`. Would that be the right approach for this?

**Comment by Gsantomaggio** [comment]
Hi @hae-anyoung 
Please have a look on: https://www.rabbitmq.com/docs/stream-core-plugin-comparison#interoperabilty

The fields`Annotations` and `MessageHeaders` are only for compatibility. This is why the fields are in readonly !

You can use `ApplicationProperties` to set the information you need.
